File: dataPreprocessing_v4/PositionFingerprint.py
from RingBufferHistory import RingbufferHistory
import ADASSegment

class PositionFingerprint:
    def __init__(self, seg, landm, odo):
        self.numSegments = seg
        self.segmentHistory = [None] * self.numSegments
        self.pos = 0

    def add_segment(self, segment):
        if self.pos < self.numSegments:
            self.segmentHistory[self.pos] = segment
            self.pos += 1

    def list_segment_history(self):
        result = ""
        for s in self.segmentHistory:
            if s != None:
                result += s.print_data() + "\n"
        return result

File: 2023.04.28_asc2kml/src/asc2kml/PositionFingerprint.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package asc2kml;

/**
 *
 * @author Q262579
 */
public class PositionFingerprint {
    public PositionFingerprint(int seg, int landm, int odo) {
        numSegments = seg;
        segmentHistory = new ADASSegment[numSegments];
        for (ADASSegment s : segmentHistory)
            s = null;
    }
    
    public void addSegment(ADASSegment segment) {
        if (pos < numSegments)
            segmentHistory[pos++] = new ADASSegment(segment);
    }
    
    public String listSegmentHistorySimple() {
        String result = new String();
        for (ADASSegment s : segmentHistory)
            if (s != null)
                result += s.listIndex() + "\n";
        return result;
    }
    
    public String listSegmentHistoryDetailed() {
        String result = new String();
        for (ADASSegment s : segmentHistory)
            if (s != null)
                result += s.listAllData() + "\n";
        return result;
    }

    private int pos = 0;
    private final int numSegments;
    // the history contains the trace of segments lading to the final (parking) position
    public ADASSegment[] segmentHistory;
    
    public Landmarks[]  landmarks;
    public VehicleOdometry[] odometries;
}
